channeldb/graph.go

```diff
@@ -405,9 +405,12 @@ func (c *ChannelGraph) ForEachNodeChannel(tx kvdb.RTx, node route.Vertex,
 	dbCallback := func(tx kvdb.RTx, e *ChannelEdgeInfo, p1,
 		p2 *ChannelEdgePolicy) error {
 
-		cachedInPolicy := NewCachedPolicy(p2)
-		cachedInPolicy.ToNodePubKey = toNodeCallback
-		cachedInPolicy.ToNodeFeatures = toNodeFeatures
+		var cachedInPolicy *CachedEdgePolicy
```

Comment:
Do we know the root cause here?

Guessing it's node that for w/e reason don't have their own self edge inserted due to past bugs where the handoff/creation wasn't reliable?

Reply:
Also shows a gap in our test coverage here as well: should add a basic test here to increase confidence in the fix.